### ISSUE TITLE
visionOS SDKs missing

### ISSUE THREAD

**Issue by noahsmartin** [actions/runner-images#10692]
### Description

The visionOS SDKs seem to have gone missing from the macos14 GitHub action runners. A couple days ago it worked, today I get this error: { platform:visionOS, id:dvtdevice-DVTiOSDevicePlaceholder-xros:placeholder, name:Any visionOS Device, error:visionOS 1.2 is not installed. To use with Xcode, first download and install the platform }

This is with Xcode 15.4, I've also tried updated to 16.0 but get the error that visionOS 2.0 is not installed. You can see that issue here: https://github.com/EmergeTools/SnapshotPreviews-iOS/actions/runs/11056651930/job/30718675120?pr=202 

Is there any way to build for visionOS is a GitHub action runner?

### Platforms affected

- [x] Azure DevOps
- [x] GitHub Actions - Standard Runners
- [x] GitHub Actions - Larger Runners

### Runner images affected

- [ ] Ubuntu 20.04
- [ ] Ubuntu 22.04
- [ ] Ubuntu 24.04
- [ ] macOS 12
- [ ] macOS 13
- [ ] macOS 13 Arm64
- [X] macOS 14
- [x] macOS 14 Arm64
- [ ] macOS 15
- [ ] macOS 15 Arm64
- [ ] Windows Server 2019
- [ ] Windows Server 2022

### Image version and build link

https://github.com/EmergeTools/SnapshotPreviews-iOS/actions/runs/11056651930/job/30718675120?pr=202

### Is it regression?

yes https://github.com/EmergeTools/SnapshotPreviews-iOS/actions/runs/10948139766

### Expected behavior

It should be installed

### Actual behavior

It is not installed

### Repro steps

Attempt to build for visionOS SDK

**Comment by sisoje** [comment]
i also have this issue, this is terrible

**Comment by erik-bershel** [comment]
Hey there!

It's kinda expected behaviour. See details here: 
- https://github.com/actions/runner-images/issues/10559

We'll keep this issue on board for better informing for some time.

**Comment by erik-bershel** [comment]
TL;DR:
- macOS-14 images do not ship `visionOS` platform

Possible workarounds:
- to use `macos-15` image
- to install visionOS in runtime with:

```bash
            sudo xcodebuild -runFirstLaunch
            sudo xcrun simctl list
            sudo xcodebuild -downloadPlatform visionOS
            sudo xcodebuild -runFirstLaunch
```

**Comment by erik-bershel** [comment]
👋 

For those tracking these changes or encountering issues for the first time, here’s some context and clarification.


As you may know, we recently faced significant challenges where many users began running out of disk space during tests and builds on `macOS`-based runners. 😞  This issue was triggered by the inclusion of `visionOS` simulators in the latest `Xcode` builds. The root cause, however, lies in our policy of providing nearly all supported minor versions of `Xcode` on each `macOS` image, leading to steadily increasing disk usage as `Apple`’s ecosystem grows.

Here’s how we’ve approached resolving the issue so far:
- **Immediate action:** We temporarily stopped pre-installing `visionOS` platform tools in `Xcode` on `macOS-14` images.
- **Strategic adjustment:** We introduced a policy to limit the number of major `Xcode` versions installed per `macOS` image, alongside reinstating `visionOS` tools for supported `Xcode` versions. This aimed to free up significant disk space. https://github.com/actions/runner-images/pull/10876

However, we’ve discovered that some projects rely on multiple major versions of `Xcode` simultaneously, making these strict measures overly restrictive. This prompted us to refine our strategy, prioritising functionality alongside disk space optimisation.

### Current Strategy 📏 

Our updated approach balances reducing installed `Xcode` versions with maintaining runner usability:

1. **`macOS-14` images:**

- Will include all minor releases of `Xcode 15` with the full platform tools suite (as before).
- Will now also include two minor releases of `Xcode 16` (excluding `visionOS` tools). These will follow a “last two” principle, with the oldest replaced by the newest as updates are released. `Beta` versions will not be included. https://github.com/actions/runner-images/pull/10962

2. **`macOS-15` images:**

- Will include all minor releases of `Xcode 16` with the full platform tools suite (as before).
- Will also include the latest available release of `Xcode 15` with the full platform tools suite. https://github.com/actions/runner-images/pull/11023

3. **Disk Space and Future Adjustments:**

- Currently, `macOS-14` images have approximately `60 GB` of free space. However, **this is not a guaranteed figure** and may decrease over time.
- When free space approaches `30 GB`, we will reassess our policies and toolsets to ensure compliance with **our promised minimum** of `14 GB` of free space.
- `macOS-15` images are not currently impacted by this issue. However, with future `Xcode` updates, policies may need to be revisited. 🤷‍♂️ 

#### Recommendations 🦮 

- If your workflows depend on `visionOS`, we recommend switching to `macOS-15` images.
- If your tasks require significant disk space, consider preparing the runner by removing unnecessary large components before starting your workflows.
